File: kaop/src/main/java/io/github/coyamo/kaop/AbsJoinPoint.java
package io.github.coyamo.kaop;

import androidx.annotation.Nullable;

import java.lang.reflect.Method;

public abstract class AbsJoinPoint {
    @Nullable
    abstract public Object proceed();

    abstract public Object getOwner();

    abstract public Method getMethod();

    abstract public Object[] getParams();
}

File: kaop/src/main/java/io/github/coyamo/kaop/AbsJoinPoint.kt
package io.github.coyamo.kaop

import java.lang.reflect.Method

/**
 * 一个JoinPoint
 * @author Coyamo
 * @date 2022/12/17 00:47
 * @version 1.0
 */
abstract class AbsJoinPoint {
    /**
     * 执行
     */
    abstract fun proceed(): Any?

    /**
     * 所属对象
     */
    abstract val owner: Any

    /**
     * 方法本体
     */
    abstract val method: Method

    /**
     * 方法的参数
     */
    abstract val params: Array<Any?>
}